plugins/Terraform-Plugin.lua

```diff
@@ -24,84 +24,99 @@
-    end
-    local request_url = ""
-    if port ~= nil then
-     	request_url = "https://" .. endpoint .. ":" .. port .. "/" .. path
-    else
-     	request_url = "https://" .. endpoint .. "/" .. path
-    end
-    
-    if request_body ~= nil then
-      response, err = request { method = method, url = request_url, headers = headers, body = json.encode(request_body) }
-    else
-      response, err = request { method = method, url = request_url, headers = headers }
-    end
-  
-    return response, err
+
+  for i = 1, length do
+    randomString = randomString .. charTable[math.random(1, #charTable)]
+  end
+  return randomString
+end
+
+function http_request(endpoint, authorization_header, port, path, request_body, method)
+  local content_type = "application/json"
+  if authorization_header ~= nil then
+      local headers = { ["Content-Type"] = content_type, ["Authorization"] = authorization_header }
+  else
+      local headers = { ["Content-Type"] = content_type }
+  end
+  local request_url = ""
+  if port ~= nil then
+     request_url = "https://" .. endpoint .. ":" .. port .. "/" .. path
+  else
+     request_url = "https://" .. endpoint .. "/" .. path
   end
   
+  if request_body ~= nil then
+    response, err = request { method = method, url = request_url, headers = headers, body = json.encode(request_body) }
+  else
+    response, err = request { method = method, url = request_url, headers = headers }
+  end
+
+  return response, err
+end
+
 function run(input)
-	if input.operation == "create" then
+if input.operation == "create" then
     local new_secret = ""
     if input.value ~= nil then
-      new_secret = input.value
+        new_secret = input.value
     else
-	    new_secret = generate_secret()
+      new_secret = generate_secret()
     end
     local new_sobject = assert(Sobject.import { name = input.name, group_id = input.group_id, obj_type = "SECRET", value = Blob.from_bytes(new_secret)})
     local resp_payload = {
-      kid      = new_sobject.kid,
-      name     = new_sobject.name,
-      group_id = new_sobject.group_id
-		}
-		return resp_payload
-    elseif input.operation == "drop" then
-        local sobject, err = Sobject { name = input.kid }
-        if sobject == nil or err ~= nil then
-            err = "[DSM SDK]: No known security object named " .. input.kid .. "."
-            return nil, err
-        end
-		assert(sobject:delete())
-		local resp_payload = {}
-        return resp_payload
+        kid      = new_sobject.kid,
+        name     = new_sobject.name,
+        group_id = new_sobject.group_id
+  }
+  return resp_payload
+  elseif input.operation == "rotate" then
+    local new_secret = ""
+    if input.value ~= nil then
+        new_secret = input.value
     else
-		err = "[DSM SDK]: No operation was specified"
-		return nil, err
+      new_secret = generate_secret()
     end
+    local sobject_old = assert(Sobject { name = "something5" })
```

Comment:
can you explain something5 here as the name?

Reply:
This needs fix

Reply:
Fixed. Hence Resolving